### PR TITLE
Make cards consistent again

### PR DESCRIPTION
**Pull request by taniki**
Based upon [Improve Cards consistency](https://github.com/opendatateam/udata/pull/1252) but with features sorted out in various PR.

This one is only about a global overlook of look and feel of cards.

## extracts

- cards in search  #1386 
- carrousel #1373 
- simplification of date range #1392

### CODE REVIEW

**Comment by noirbizarre** [comment]
I think you need to rebase on the current master. The changelog seems to indicate that you are working a an old version of the master

**Comment by taniki** [comment]
@noirbizarre I have conflicts with your two last merges. I have integrated the second one but overrided part of the first one.

**Comment by abulte** [comment]
This is awesome 🎉 👏 

Some feedback after a tour on `demo`:

- On a meta level, I'm worried we're giving too much prominence (space wise) to reuses on HP. We had a number of talks about the fact that maybe reuses are not as useful / pertinent as they were and we definitely have less and less of them. Maybe we need to tackle this when we refactor the carousel. We should probably have a meeting with everyone interested at Etalab on this topic too.

- Misalignment on tabs on HP

capture d ecran 2018-02-02 10 34 00

- Carousel and reuses cards view for a topic is the "old one", we should probably switch to the view from HP

capture d ecran 2018-02-02 10 41 24

- Spacing (top) on tabs of an org page could be better IMHO. Also, why not 3 datasets in a row like on HP? It works well.

capture d ecran 2018-02-02 10 59 19

- Strange artifact on bottom right corn of timeline's cards

capture d ecran 2018-02-02 10 55 19

- Organization card title is not clickable (tested on timeline) http://demo.data.gouv.fr/fr/users/nathalie-vernusprost/

- Grid placement problems on Safari :-/

capture d ecran 2018-02-02 10 58 11

capture d ecran 2018-02-02 10 59 19

capture d ecran 2018-02-02 11 01 09

File: js/components/dataset/card.vue [hunk 1]
```diff
@@ -3,70 +3,52 @@
...
+            <li v-tooltip :title="_('Resources count')">
+                <span class="fa fa-files-o fa-fw"></span>
+                {{ dataset.resources.length  }}
             </li>
-            <li v-if="dataset.metrics">
-                <a class="btn btn-xs" v-tooltip tooltip-placement="top" :title="_('Reuses')">
-                    <span class="fa fa-retweet fa-fw"></span>
-                    {{ dataset.metrics.reuses || 0 }}
-                </a>
+            <li v-if="dataset.spatial && dataset.spatial.zones && dataset.spatial.zones.length > 0" v-tooltip :title="_('Territorial coverage')">
```
Comment (abulte) [inline js/components/dataset/card.vue]: no need for `> 0` I think

File: less/udata/resource.less [hunk 2]
```diff
@@ -55,11 +55,16 @@
         }
 
         &.expanded {
-            border-bottom-color: @base-color;
+            border-bottom-color: #fafafa;
```
Comment (abulte) [inline less/udata/resource.less]: variable?

File: less/udata/resource.less [hunk 3]
```diff
@@ -55,11 +55,16 @@
...
         }
 
         &.body-only {
-            border-top-color: @base-color;
+            margin-top: -@vspacing;
+            border-color: #fafafa;
```
Comment (abulte) [inline less/udata/resource.less]: variable?

File: udata/templates/discussion/threads.html [hunk 4]
```diff
@@ -9,55 +9,57 @@
...
-            <form role="form" class="clearfix animated"
-                data-api-url="{{ url_for('api.discussions', for=subject.id) }}"
-                data-subject-class="{{ subject.__class__.__name__ }}"
-                data-subject-id="{{ subject.id }}">
-                <div class="form-group">
-                    <label for="title-new-discussion">{{ _('Title') }}</label>
-                    <input type="text" id="title-new-discussion" name="title" class="form-control" required />
-                    <label for="comment-new-discussion">{{ _('Comment') }}</label>
-                    <textarea id="comment-new-discussion" name="comment" class="form-control" rows="3" required></textarea>
+    <div class="{{ threadclass }} resources-list smaller">
```
Comment (abulte) [inline udata/templates/discussion/threads.html]: Is this supposed to be in this PR? Seems unrelated to cards.

File: udata/templates/reuse/card.html [hunk 5]
```diff
@@ -1,55 +1,62 @@
...
+    <div class="card__cover">
+        <a href="{{ url_for('reuses.show', reuse=reuse) }}">
+            <img class="media-object img-responsive" alt="{{ reuse.title }}"
+                src="{{ reuse.image|placeholder('reuse') }}">
+        </a>
+    </div>
+    {% else %}
     <a class="card-logo" href="{{ url_for('reuses.show', reuse=reuse) }}">
-        <img alt="{{ reuse.title }}"
+        <img alt=""
```
Comment (abulte) [inline udata/templates/reuse/card.html]: why u no alt?

File: udata/templates/territories/card.html [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{% cache cache_duration, 'territory-search-result', territory.id, g.lang_code %}
```
Comment (abulte) [inline udata/templates/territories/card.html]: A search on `territories/card.html` yields nothing, both in current code base and this PR. Is this used?

File: udata/templates/user/datasets.html [hunk 7]
```diff
@@ -32,12 +32,12 @@
 
 	<div class="row">
 	    <div class="col-xs-12">
-	
+            <p class="text-center lead">
+            {{ _('%(user)s has not yet published dataset', user=user.fullname) }}
```
Comment (abulte) [inline udata/templates/user/datasets.html]: s/published dataset/published any dataset/

File: udata/templates/user/reuses.html [hunk 8]
```diff
@@ -31,12 +31,12 @@
 
 	<div class="row">
 	    <div class="col-xs-12">
-	
+            <p class="text-center lead">
+            {{ _('%(user)s has not yet published resuse', user=user.fullname) }}
```
Comment (abulte) [inline udata/templates/user/reuses.html]: s/has not yet published resuse/has not yet published any reuse/

File: js/components/dataset/card.vue (hunk 1 above)
Comment (abulte) [reply js/components/dataset/card.vue]: This does not seem to display anything: capture d ecran 2018-02-02 11 54 15

File: udata/templates/dataset/card.html [hunk 9]
```diff
@@ -17,75 +17,51 @@
...
-            {% if dataset.spatial and dataset.spatial.granularity %}
-            <li>
-                <a class="btn btn-xs" v-tooltip title="{{ _('Territorial coverage granularity') }}">
-                    <span class="fa fa-bullseye fa-fw"></span>
-                    {{ dataset.spatial.granularity_label }}
-                </a>
+            {% if dataset.spatial and dataset.spatial.territories %}
+            <li v-tooltip title="{{ _('Territorial coverage') }}">
+                <span class="fa fa-map-marker fa-fw"></span>
+                {{ dataset.spatial.top_label }}
```
Comment (abulte) [inline udata/templates/dataset/card.html]: I could not find any place where this is working. It should display "France" on a bunch of datasets on the HP if I'm not mistaken.

File: udata/templates/territories/card.html (hunk 6 above)
Comment (taniki) [reply udata/templates/territories/card.html]: https://github.com/dtc-innovation/udata/blob/093608c4a30d75b64941349a7114852c20a96e92/udata/templates/search.html

In the new search results apparently (:

Comment (abulte) [reply udata/templates/territories/card.html]: To be moved to another PR then ;-)

File: udata/templates/discussion/threads.html (hunk 4 above)
Comment (taniki) [reply udata/templates/discussion/threads.html]: conflict resolution

File: udata/templates/dataset/card.html (hunk 9 above)
Comment (taniki) [reply udata/templates/dataset/card.html]: blind guess: this is a messed up by some parallel workaround on "territories". I have put back what have been used as a condition to show top zone or not.

**Comment by abulte** [comment]
One last thing and after that I think LGTM 🎉 

Misalignment topic page tab (was OK before review fixes)
capture d ecran 2018-02-08 09 14 19

**Comment by noirbizarre** [comment]
Other bug, the add button on dataset page:
screenshot-demo data gouv fr-2018 02 08-10-54-31

**Comment by taniki** [comment]
@noirbizarre not even sure about what it should look like. A card or are we keeping the former presentation?

**Comment by abulte** [comment]
There are problems with some slides on the carousel with the default theme:

capture d ecran 2018-02-08 15 21 38
capture d ecran 2018-02-08 15 23 26

**Comment by abulte** [comment]
I wonder if we should have some different, more neutral colors for cards footer in the default theme. BTW, we agreed some time ago that the cards footer colors should be configurable, is this the case?

**Comment by noirbizarre** [comment]
Former presentation is fine to be, but this is currently broken.

Incoherent on default theme too:

screenshot-demo data gouv fr-2018 02 08-15-33-56

**Comment by noirbizarre** [comment]
Certification stamp seems to be missing now:

| Before | After |
|-----------|--------|
| screenshot-data xps-2018 02 08-15-44-26 | screenshot-demo data gouv fr-2018 02 08-15-44-54 |

**Comment by abulte** [comment]
There is a slight problem with tooltips on texts that overflow a lot (this is the tooltip of the left card):

capture d ecran 2018-02-08 17 26 14

**Comment by abulte** [comment]
Slight problem between breakpoints on the admin (~1200px wide here):

capture d ecran 2018-02-08 17 38 57

https://demo.data.gouv.fr/fr/admin/editorial/

**Comment by abulte** [comment]
Problem with dataset card on reuse page in the admin:

capture d ecran 2018-02-08 17 46 02

https://demo.data.gouv.fr/fr/admin/reuse/596f5fd5c751df02a36e4726/

**Comment by taniki** [comment]
The two previous bugs have been fixed. Can someone get a detailed look at the very last commit? I have change a stuff in the api and need an impact assesment

**Comment by taniki** [comment]
screen shot 2018-02-09 at 15 57 48

**Comment by abulte** [comment]
@taniki regarding your question, it might be better to add the fields you need to `dataset_ref_fields` instead of of getting all the fields. Which ones do you need?

**Comment by noirbizarre** [comment]
Reuse cards from companies display `None` as author:
screenshot-demo data gouv fr-2018 02 12-17-21-21

**Comment by abulte** [comment]
Problems on user and org cards when creating a dataset:

capture d ecran 2018-02-13 16 03 07
capture d ecran 2018-02-13 16 03 17

**Comment by abulte** [comment]
Problem on topic admin view:

https://demo.data.gouv.fr/fr/admin/topic/53552f47c751df3ceca6f823

capture d ecran 2018-02-13 16 07 55

**Comment by abulte** [comment]
Misaligned close icon on topic dataset edition (clearly not critical):

capture d ecran 2018-02-13 16 10 02

**Comment by abulte** [comment]
Still a small glitch on user card when creating a dataset (can be fixed later):

capture d ecran 2018-02-14 09 26 29
